pkg/doc/signature/proof/jws.go

```diff
@@ -129,6 +129,8 @@ func prepareDocumentForJWS(suite signatureSuite, jsonldObject map[string]interfa
 	doc := GetCopyWithoutProof(jsonldObject)
 
 	if suite.CompactProof() {
+		opts = append(opts, jsonld.WithDocumentLoaderCache(jsonldCache))
```

Comment:
@troyronda Here we apply extra cached contexts (e.g. security v2 context) and still use input opts. Input opts can contain additional caches or custom context loader with defined cached context maps. All of them (cache) will be utilized when making JSON-LD operations like canonization or compaction.